### PR TITLE
Ft dsk 972 venue features api

### PR DESCRIPTION
**Pull request by RafaelDelboni**

### CODE REVIEW

File: src/resources/Resource.js [hunk 1]
```diff
@@ -117,11 +124,10 @@ export default class Resource {
       throw new DeskbookersError(msg)
 
     // If "errors" exists in response
-    } else if (errors) {
-      for (let error in errors) {
-        const msg = errors[error].title || errors[error].detail
-        throw new DeskbookersError(msg)
-      }
+  } else if (errors) {
```
Comment (sabarasaba) [inline src/resources/Resource.js]: ❤️

File: docs/features.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+{
+  "status": "ok"
+}
+```
+
+## `listByVenue(venueId)`
+List all active features for a given `venueId`.
+
+```js
+const data = await deskbookers.feature.listByVenue(1)
```
Comment (sabarasaba) [inline docs/features.md]: @RafaelDelboni can we also return the endDate for each feature ?

File: docs/features.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+```js
+  const data = await deskbookers.feature.checkFeatureByVenue(
+    1,
+    'regularCustumerDiscounts'
+  )
+```
+#### Example response
+```json
+{
```
Comment (vdvleon) [inline docs/features.md]: This SDK method should return `true` or `false` not an object with `{enabled}`.

File: src/resources/Resource.js [hunk 4]
```diff
@@ -37,46 +37,53 @@ export default class Resource {
...
-      options.body = queryStr
+    if (!isEmpty(body)) {
+      options.headers['Content-Type'] = 'application/json'
+      options.body = phpJsonEncode(body)
+      options.rawBody = true
+      args = body
       url = `${this.apiUrl}/${pathFixed}`
     } else {
-      url = `${this.apiUrl}/${pathFixed}?${queryStr}`
+      const shouldEncodeArgs = options.method === 'POST'
```
Comment (vdvleon) [inline src/resources/Resource.js]: Should also work for PUT

File: docs/features.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+]
+```
+
+## `checkFeatureByVenue(venueId, featureName)`
+List all active features for a given `venueId` and `featureName`.
+
+```js
+  const data = await deskbookers.feature.checkFeatureByVenue(
+    1,
+    'regularCustumerDiscounts'
```
Comment (vdvleon) [inline docs/features.md]: Typo: `regularCustomerDIscount`

**Pull request merged**